main/testkit/src/testkit/MillTestkit.scala

```diff
@@ -72,7 +72,7 @@ trait MillTestKit {
       debugEnabled: Boolean = false,
       extraPathEnd: Seq[String] = Seq.empty
   )(implicit fullName: sourcecode.FullName) {
-    val outPath = getOutPath(testPath) / extraPathEnd
+    val outPath = module.millSourcePath / "out" / testPath / extraPathEnd
```

Comment:
Why you replaced the `getOutPath` call with a rather hard-coded path?

Reply:
So the issue here is that in real life (not in tests) we have the `millSourcePath` pointing to the top project folder and then we have a subfolder `out`.
So in real life, the outPath (or workspace) is nested into the worksources.
Tests need to be the same and my change is making it closer to real life.

Reply:
I'm not sold by this argumentation. There _is_ the option of overriding `def getOutPath` to customize it to your needs. If there is a indirection, there is a potential need for it. Also, in other tests we create "real life" test setups by copying the `worksources` into the `workspace`. to the contrary, in tests we need more control over the world under test, so we can actually separate preconditions, assumptions and results.

Reply:
Outside of tests, this PR relies on doing a docker build in `T.workspace` assuming all files we are pointing to in the Dockerfile are in `T.workspace`. Now this does not work if out path is not a sub folder of `T.workspace`. Are we in agreement that out path needs to be a sub folder of `T.workspace`?

Reply:
> Outside of tests, this PR relies on doing a docker build in `T.workspace` assuming all files we are pointing to in the Dockerfile are in `T.workspace`. Now this does not work if out path is not a sub folder of `T.workspace`. Are we in agreement that out path needs to be a sub folder of `T.workspace`?

I think, this is an assumption that may not hold `true` in all conditions. IMHO, we don't want this hard requirement. Also, I can't exactly say, how this behaves for foreign modules. (mill-docker  will probably fail because either `T.workspace` is no longer the dir containing `build.sc` or the `out` location changes.)

Maybe, using `T.workspace` is not the right thing then. You could introduce a `dockerRoot` target, which can be customized by the user. The default could be just `millSourcePath`.

After all, I think the merge of this PR was premature and changing core Mill to get a change in a contrib package through without final approval is an unwanted behavior. At least, I don't like it.

Reply:
I'm reverting this and creating a new PR with that suggestion